src/plots/layout_attributes.js

```diff
@@ -45,13 +45,17 @@ module.exports = {
         description: 'Sets the title font.'
     }),
     autosize: {
-        valType: 'enumerated',
+        valType: 'boolean',
         role: 'info',
-        // TODO: better handling of 'initial'
-        values: [true, false, 'initial'],
+        dflt: false,
         description: [
-            'Determines whether or not the dimensions of the figure are',
-            'computed as a function of the display size.'
+            'Determines whether or not a layout width or height',
+            'that has been left undefined by the user',
+            'is initialized on each relayout.',
+
```

Comment:
Yes, this wording could be improved. The most important points I wanted to make in the description were that:
- if the user supplies values of `layout.width` and/or `layout.height`, `plotAutoSize` doesn't change them.
- if `layout.autosize` is false, `plotAutosize` is called once.
- if `layout.autosize` is true, `plotAutosize` is called on each relayout.